Sources/TuistLoader/Utils/SetupLocator.swift

```diff
@@ -0,0 +1,30 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+public protocol SetupLocating {
+    /// It traverses up the directory hierarchy until it finds a `Setup.swift` file.
+    /// - Parameter path: Path from where to do the lookup.
+    func locate(at path: AbsolutePath) -> AbsolutePath?
+}
+
+public final class SetupLocator: SetupLocating {
+    let rootDirectoryLocator: RootDirectoryLocating
+
+    public init(rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator()) {
+        self.rootDirectoryLocator = rootDirectoryLocator
+    }
+
+    public func locate(at path: AbsolutePath) -> AbsolutePath? {
+        let manfiestPath = path.appending(component: Manifest.setup.fileName(path))
```

Comment:
- typo in `manifestPath`
- If you put this along with `Config` locating into `ManifestFilesLocator`, you could even reuse some of the logic.
- Is there a reason you do not use `rootDirectoryLocator`? Seems like a good use case for it